src/wallet/rpc/spend.cpp

```diff
@@ -548,6 +549,32 @@ void FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& fee_out,
         }
     }
 
+    if (options.exists("input_weights")) {
+        for (const UniValue& input : options["input_weights"].get_array().getValues()) {
+            uint256 txid = ParseHashO(input, "txid");
+
+            const UniValue& vout_v = find_value(input, "vout");
+            if (!vout_v.isNum()) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, missing vout key");
+            }
+            int vout = vout_v.get_int();
+            if (vout < 0) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, vout cannot be negative");
+            }
+
+            const UniValue& weight_v = find_value(input, "weight");
+            if (!weight_v.isNum()) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, missing weight key");
+            }
+            int64_t weight = weight_v.get_int64();
+            if (weight < 40 * WITNESS_SCALE_FACTOR) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, weight cannot be less than 160 (40 bytes (size of outpoint + sequence) * 4 (witness scaling factor))");
+            }
```

Comment:
Did you consider having an upper bound for weights as well? e.g. based on `MAX_STANDARD_SCRIPTSIG_SIZE * WITNESS_SCALE_FACTOR` or even `MAX_STANDARD_TX_WEIGHT`?